drivers/volume/aws/aws.go

```diff
@@ -368,9 +380,15 @@ func (a *aws) CancelBackup(backup *storkapi.ApplicationBackup) error {
 }
 
 func (a *aws) DeleteBackup(backup *storkapi.ApplicationBackup) (bool, error) {
-	if a.client == nil {
-		if err := a.Init(nil); err != nil {
-			return true, err
+	var client *ec2.EC2
+	// if backuplocation has creds wrt the cluster, need to use that
+	client = a.getAWSClientFromBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
+	if client == nil {
```

Comment:
Suggestion: Can we have even calling `init()` inside `getAWSClientFromBackupLocation()` that way we don't have to repeat and any other calling will get the needed handle.